packages/components/src/edit-in-place-control/index.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useRef, useEffect } from '@wordpress/element';
+import { ENTER, ESCAPE } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+
+function EditInPlaceControl( { label = '', onClick = noop, onUpdate = noop } ) {
+	const [ edit, setEdit ] = useState( false );
+	const [ value, setValue ] = useState( label );
+	const prevValue = useRef();
+
+	const inputRef = useRef();
+	const buttonRef = useRef();
+
+	useEffect( () => {
+		prevValue.current = value;
+		if ( edit ) {
+			inputRef.current.focus();
+			inputRef.current.select();
+		} else {
+			buttonRef.current.focus();
+		}
+	}, [ edit ] );
+
+	return (
+		<>
+			{ ! edit && (
+				<Button
+					ref={ buttonRef }
+					className="components-edit-in-place-control__label"
+					onClick={ () => {
+						setEdit( true );
+						onClick();
+					} }
+				>
+					{ value }
+				</Button>
+			) }
+			{ edit && (
+				<input
+					ref={ inputRef }
+					className="components-edit-in-place-control__input"
+					value={ value }
+					onChange={ ( event ) => {
+						setValue( event.target.value );
+					} }
+					onBlur={ () => {
+						setEdit( false );
+						onUpdate( value );
```

Comment:
I guess it could be the idea that the display value of the edit in place component is _updated_, not that the input is submitted, but I see your point.

Reply:
As for now, I have left `onUpdate`. We can continue the discussion in the PR where the logic of this component has been moved https://github.com/WordPress/gutenberg/pull/28924